src/unix/process.c

```diff
@@ -199,8 +202,24 @@ static int uv__process_init_stdio(uv_stdio_container_t* container, int fds[2]) {
     assert(container->data.stream != NULL);
     if (container->data.stream->type != UV_NAMED_PIPE)
       return UV_EINVAL;
-    else
-      return uv_socketpair(SOCK_STREAM, 0, fds, 0, 0);
+    else {
+      ret = uv_socketpair(SOCK_STREAM, 0, fds, 0, 0);
+
+#ifdef __APPLE__
+    if (ret == 0) {
+      for (int i = 0; i < 2; i++) {
+        for (int j = 0; j < 2; j++) {
+          ret = setsockopt(fds[i], SOL_SOCKET, options[j], &size, sizeof(size));
```

Comment:
I'd ignore the errors here. If we fail to increase the buffer size (not sure why that would happen TBH) so be it

Reply:
```C
#ifdef __APPLE__
    if (ret == 0)
      for (int i = 0; i < 2; i++)
        for (int j = 0; j < 2; j++)
          setsockopt(fds[i], SOL_SOCKET, options[j], &size, sizeof(size));
#endif
```
Would it be acceptable to make this change?

Reply:
I plan to modify the code as follows.

```C
ret = uv_socketpair(SOCK_STREAM, 0, fds, 0, 0);

if (ret == 0)
  for (i = 0; i < 2; i++) {
    setsockopt(fds[i], SOL_SOCKET, SO_RCVBUF, &size, sizeof(size));
    setsockopt(fds[i], SOL_SOCKET, SO_SNDBUF, &size, sizeof(size));
  }

return ret;
```

Additionally, I noticed that the test fails on Unix if the Apple guard is not applied.  
On Unix, the buffer size is automatically set to twice the specified value.  
So, if we set it to `64 * 1024`, it becomes 128KB on Unix.

Wouldn't it be better to apply the Apple guard in this case?

Reply:
Since we have decided to ignore failures when increasing the buffer size,  
should I remove the buffer size verification test that I added?

Reply:
Humm, I suspect we don't want to enlarge the size of its already 64k indeed.